ant-node/src/bin/antnode/main.rs

```diff
@@ -448,24 +480,39 @@ You can check your reward balance by running:
         );
     }
 
-    // If auto-restart delay is specified, spawn a task to trigger restart after the delay
-    if let Some(delay_secs) = auto_restart_delay {
+    #[cfg(unix)]
+    {
         let ctrl_tx_restart = ctrl_tx.clone();
+        let running_node_clone = running_node.clone();
         tokio::spawn(async move {
-            let delay = Duration::from_secs(delay_secs);
-            info!("Auto-restart enabled: node will restart in {delay:?}");
-            println!("Auto-restart enabled: node will restart in {delay:?}");
-            sleep(delay).await;
-            info!("Auto-restart delay elapsed, triggering node restart");
-            println!("Auto-restart delay elapsed, triggering node restart");
-            if let Err(err) = ctrl_tx_restart
-                .send(NodeCtrl::Restart {
-                    delay: Duration::from_secs(1),
-                    retain_peer_id: true,
-                })
-                .await
-            {
-                error!("Failed to send auto-restart command: {err}");
+            loop {
+                // 72 hours (259200 seconds) with ±5% randomization to prevent simultaneous upgrades
+                let base_delay = 259200;
+                let variance = rand::thread_rng().gen_range(-12960..=12960);
+                let delay_secs = (base_delay + variance) as u64;
+                sleep(Duration::from_secs(delay_secs)).await;
+                match upgrade::perform_upgrade().await {
+                    Ok(()) => {
+                        info!("Upgrade successful. Triggering restart...");
+
+                        let delay = calculate_restart_delay(&running_node_clone).await;
+                        info!("Calculated restart delay: {delay:?}");
+
+                        if let Err(err) = ctrl_tx_restart
+                            .send(NodeCtrl::Restart {
+                                delay,
+                                retain_peer_id: true,
+                            })
+                            .await
```

Comment:
The `delay` value is sent via `NodeCtrl::Restart` and the sleep is carried out within the code at line 520+ inside the same file. That loop also captures `NodeCtrl::Stop` which is used to stop nodes using Ctrl+c or during any critical error. It would be nice to maybe remove the `delay` field from the `NodeCtrl::Restart` variant and instead sleep directly inside this `upgrade checking loop` before sending out the `NodeCtrl::Restart`